apis/k8ssandra/v1alpha1/k8ssandracluster_types.go

```diff
@@ -422,6 +422,13 @@ type DatacenterOptions struct {
 
 	// The k8s service account to use for the Cassandra pods
 	ServiceAccount string `json:"serviceAccount,omitempty"`
+
+	// DatacenterName allows to override the name of the Cassandra datacenter. Kubernetes objects will be named after a sanitized version of it if set, and if not metadata.name. In Cassandra the DC name will be overridden by this value.
+	// It may generate some confusion as objects created for the DC will have a different name than the CasandraDatacenter object itself.
```

Comment:
> does this mean that objects created for the DC will use DatacenterName, but the CassandraDatacenter still uses metadata.name?

Yes, and that's a decision we made a while ago for cluster name overrides in K8ssandraCluster objects, which we're trying to be consistent with.
While this may seem confusing, there's another way to view this which is that clusters aren't referenced by their Kubernetes technical name but rather the Cassandra cluster name. Naming the objects accordingly makes it easier to spot which pod/configmap relates to what cluster.
Not an easy call, and opinions surely differ regarding what the "right" decision should be here.
Note that overrides aren't common and that sane operators would use a K8ssandraCluster name that is a sanitized version of the override, hopefully.